lib/spack/spack/cmd/log.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import shutil
+import tempfile
+import time
+
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.environment as ev
+import spack.util.compression as compression
+from spack.cmd.common import arguments
+
+description = "print out logs for packages"
+section = "basic"
+level = "long"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        "--continuous", action="store_true", help="for stage logs, poll for new content"
+    )
+
+    arguments.add_common_arguments(subparser, ["spec"])
+
+
+def dump_build_log(path, continuous):
+    with open(path, "r") as fstream:
```

Comment:
I would open in binary mode so you don't have to worry about potentially invalid unicode in logs, which happens every now and then.